src/App.tsx

```diff
@@ -17,18 +68,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                todoStatus={todoStatus}
+                selectStatus={setTodoStatus}
+                query={query}
+                setQuery={setQuery}
+              />
             </div>
 
-            <div className="block">
-              <Loader />
-              <TodoList />
-            </div>
+            {!dataIsLoaded
+              ? (
+                <div className="block">
+                  <Loader />
+                </div>
+              ) : (
+                <div className="block">
+                  <TodoList
+                    todos={filterTodos()}
+                    selectedTodoId={selectedTodoId}
+                    selectTodo={handleTodoSelect}
+                  />
+                </div>
```

Comment:
You are a little bit repetitive here, both `Loader` and `TodoList` are inside `div.block`, so you could move conditional rendering inside of this block